components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -0,0 +1,571 @@
+        fn report(&self, apply_res: ApplyRes) {
+            let _ = self.sender.send(apply_res);
+        }
+    }
+
+    fn new_split_req(key: &[u8], id: u64, children: Vec<u64>) -> SplitRequest {
+        let mut req = SplitRequest::default();
+        req.set_split_key(key.to_vec());
+        req.set_new_region_id(id);
+        req.set_new_peer_ids(children);
+        req
+    }
+
+    fn assert_split(
+        apply: &mut Apply<engine_test::kv::KvTestEngine, MockReporter>,
+        factory: &Arc<TestTabletFactoryV2>,
+        region_to_split: &Region,
+        right_derived: bool,
+        new_region_ids: Vec<u64>,
+        split_keys: Vec<Vec<u8>>,
+        children_peers: Vec<Vec<u64>>,
+        log_index: u64,
+        region_boundries: Vec<(&[u8], &[u8])>,
+    ) -> HashMap<u64, Region> {
+        let mut splits = BatchSplitRequest::default();
+        splits.set_right_derive(right_derived);
+
+        for ((new_region_id, children), split_key) in new_region_ids
+            .into_iter()
+            .zip(children_peers.clone())
+            .zip(split_keys)
+        {
+            splits
+                .mut_requests()
+                .push(new_split_req(&split_key, new_region_id, children));
+        }
+
+        let mut req = AdminRequest::default();
+        req.set_splits(splits);
+
+        // Exec batch split
+        let (resp, apply_res) = apply.apply_batch_split(&req, log_index).unwrap();
+
+        let regions = resp.get_splits().get_regions();
+        assert!(regions.len() == region_boundries.len());
+
+        let mut epoch = region_to_split.get_region_epoch().clone();
+        epoch.version += children_peers.len() as u64;
+
+        let mut child_idx = 0;
+        for (i, region) in regions.iter().enumerate() {
+            assert_eq!(region.get_start_key().to_vec(), region_boundries[i].0);
+            assert_eq!(region.get_end_key().to_vec(), region_boundries[i].1);
+            assert_eq!(*region.get_region_epoch(), epoch);
+
+            if region.id == region_to_split.id {
+                let state = apply.region_state();
+                assert_eq!(state.tablet_index, log_index);
+                assert_eq!(state.get_region(), region);
+                let tablet_path = factory.tablet_path(region.id, log_index);
+                assert!(factory.exists_raw(&tablet_path));
+            } else {
+                assert_eq! {
+                    region.get_peers().iter().map(|peer| peer.id).collect::<Vec<_>>(),
+                    children_peers[child_idx]
+                }
+                child_idx += 1;
+
+                let tablet_path =
+                    factory.tablet_path_with_prefix(SPLIT_PREFIX, region.id, RAFT_INIT_LOG_INDEX);
+                assert!(factory.exists_raw(&tablet_path));
+            }
+        }
+
+        regions
+            .iter()
+            .map(|region| (region.id, region.clone()))
+            .collect()
+    }
+
+    #[test]
+    fn test_split() {
```

Comment:
Don't write `assert_split` again and again, should use table test instead.

And derived_index should also be tested.